commit d99609a3eb8bc96c3af841fd35294a679e0fea7f
Author: Florian Weimer <fweimer@redhat.com>
Date:   Wed Sep 6 18:37:21 2023 +0200

    manual: Fix ld.so diagnostics menu/section structure
    
    And shorten the section/node names a bit, so that the menu
    entries become easier to read.
    
    Texinfo 6.5 fails to process the previous structure:
    
    ./dynlink.texi:56: warning: node `Dynamic Linker Introspection' is
      next for `Dynamic Linker Diagnostics' in sectioning but not in menu
    ./dynlink.texi:56: warning: node up `Dynamic Linker Diagnostics'
      in menu `Dynamic Linker Invocation' and
      in sectioning `Dynamic Linker' differ
    ./dynlink.texi:1: node `Dynamic Linker' lacks menu item for
      `Dynamic Linker Diagnostics' despite being its Up target
    ./dynlink.texi:226: warning: node prev `Dynamic Linker Introspection' in menu `Dynamic Linker Invocation'
      and in sectioning `Dynamic Linker Diagnostics' differ
    
    Texinfo 7.0.2 does not report an error.
    
    This fixes commit f21962ddfc8bb23e92597da1f98e313dbde11cc1
    ("manual: Document ld.so --list-diagnostics output").
    
    Reviewed-by: Szabolcs Nagy <szabolcs.nagy@arm.com>

diff --git a/manual/dynlink.texi b/manual/dynlink.texi
index df41c56bfc..06a6c15533 100644
--- a/manual/dynlink.texi
+++ b/manual/dynlink.texi
@@ -18,6 +18,7 @@ Dynamic linkers are sometimes called @dfn{dynamic loaders}.
 @end menu
 
 @node Dynamic Linker Invocation
+@section Dynamic Linker Invocation
 
 @cindex program interpreter
 When a dynamically linked program starts, the operating system
@@ -54,7 +55,7 @@ Print system diagnostic information in a machine-readable format.
 @end menu
 
 @node Dynamic Linker Diagnostics
-@section Dynamic Linker Diagnostics
+@subsection Dynamic Linker Diagnostics
 @cindex diagnostics (dynamic linker)
 
 The @samp{ld.so --list-diagnostics} produces machine-readable
@@ -90,7 +91,13 @@ The built-in dynamic linker search path contains two directories,
 @code{/lib64} and @code{/usr/lib64}.
 @end table
 
-@subsection Dynamic Linker Diagnostics Output Format
+@menu
+* Dynamic Linker Diagnostics Format::  Format of ld.so output.
+* Dynamic Linker Diagnostics Values::  Data contain in ld.so output.
+@end menu
+
+@node Dynamic Linker Diagnostics Format
+@subsubsection Dynamic Linker Diagnostics Format
 
 As seen above, diagnostic lines assign values (integers or strings) to a
 sequence of labeled subscripts, separated by @samp{.}.  Some subscripts
@@ -132,7 +139,8 @@ subscript = label [index]
 line = subscript *("." subscript) "=" value
 @end smallexample
 
-@subsection Dynamic Linker Diagnostics Values
+@node Dynamic Linker Diagnostics Values
+@subsubsection Dynamic Linker Diagnostics Values
 
 As mentioned above, the set of diagnostics may change between
 @theglibc{} releases.  Nevertheless, the following table documents a few
